packages/driver/src/cross-origin/communicator.ts

```diff
@@ -19,42 +19,41 @@ const SNAPSHOT_EVENT_PREFIX = `${CROSS_ORIGIN_PREFIX}snapshot:`
  * @param event the name of the event to be promisified.
  * @param specBridgeName the name of the spec bridge receiving the event.
  * @param communicator the communicator that is sending the message
+ * @param timeout - in ms, if the promise does not complete during this timeout, fail the promise.
```

Comment:
added a timeout parameter since some of the automation/backend calls take longer than a second.